src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,216 @@
+    switch (variant) {
+      case 'primary':
+      default:
+        if (disabled) {
+          return variantColor ? theme.colors[variantColor][200] : theme.colors.primary[200]; //TODO: exception color used in design
+        }
+        return variantColor ? theme.colors[variantColor][800] : theme.colors.primary[800]; //in design, "Exception" white is used.
+      case 'secondary':
+        if (disabled) {
+          return variantColor ? theme.colors[variantColor][100] : theme.colors.shade[100];
+        }
+        return variantColor ? theme.colors[variantColor][200] : theme.colors.primary[200];
+      case 'tertiary':
+        return 'transparent';
+    }
+  },
+  border({ variantColor, variant, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+      case 'tertiary':
+      default:
+        return null;
+      case 'secondary':
+        if (disabled) {
+          return variantColor
+            ? `${themeHelper.px(0.125)} solid ${theme.colors[variantColor][300]}`
+            : `${themeHelper.px(0.125)} solid ${theme.colors.shade[300]}`;
+        }
+        return variantColor
+          ? `${themeHelper.px(0.125)} solid ${theme.colors[variantColor][600]}`
+          : `${themeHelper.px(0.125)} solid ${theme.colors.primary[600]}`;
+    }
+  },
+  underlayColor({ variantColor, variant, theme }) {
+    switch (variant) {
+      case 'primary':
+      default:
+        return variantColor ? theme.colors[variantColor][900] : theme.colors.primary[900]; //in design, "Exception" white is used.
+      case 'secondary':
+        return variantColor ? theme.colors[variantColor][500] : theme.colors.primary[500];
+      case 'tertiary':
+        return variantColor ? theme.colors[variantColor][500] : theme.colors.primary[500];
+    }
+  },
+  padding({ size, theme, text }) {
+    switch (size) {
+      case 'xsmall':
+        if (!text) {
+          return `${theme.spacings.xxs}`;
+        }
+        return `${theme.spacings.xxs} ${theme.spacings.s}`;
+      case 'small':
+        if (!text) {
+          return `${themeHelper.px(0.75)}`;
+        }
+        return `5px ${theme.spacings.l}`; //TODO: vertical padding is 5px in Figma, unit is 8
+      case 'medium':
+      default:
+        if (!text) {
+          return `${theme.spacings.s}`;
+        }
+        return `${theme.spacings.s} ${theme.spacings.xxl}`;
+      case 'large':
+        if (!text) {
+          return `${themeHelper.px(1.25)}`;
+        }
+        return `${themeHelper.px(1.25)} ${theme.spacings.xxl} `;
+    }
+  },
+  iconSize(size) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+      case 'large':
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)((props) => {
+  return `
+    flex-direction: row;
+    align-items: center;
+    align-self: ${props.align};
```

Comment:
Can't we push this to the ```Flex``` component?

Reply:
Once the Flex component is merged in the master. I will add it here. I haven't checked it yet.